src/codegen.cpp

```diff
@@ -9355,10 +9355,16 @@ void add_cc_args(CodeGen *g, ZigList<const char *> &args, const char *out_dep_pa
                 args.append(g->zig_target->llvm_cpu_name);
             }
             if (g->zig_target->llvm_cpu_features != nullptr) {
-                args.append("-Xclang");
-                args.append("-target-feature");
-                args.append("-Xclang");
-                args.append(g->zig_target->llvm_cpu_features);
+                // https://github.com/ziglang/zig/issues/5017
+                char *flags = buf_ptr(buf_create_from_str(g->zig_target->llvm_cpu_features));
+                char *flag = strtok(flags, ",");
```

Comment:
```suggestion
                char *saveptr;
                char *flag = strtok_r(flags, ",", &saveptr);
```
Using the non-threadsafe global-modifying `strtok` is a bit of a footgun. `strtok_r` is safer and, IMO, easier to understand.

Reply:
There's also `SplitIterator` in util.cpp

Reply:
Thanks for the suggestions, I've decided to use `strtok_r` in the end.

Reply:
It looks like `strtok_r` is POSIX only, I've switched to `SplitIterator`.